Goose.API/Services/issues/IssueDetailedService.cs

```diff
@@ -2,95 +2,90 @@
         }
 
         private async Task<IssueDTODetailed> CreateDtoFromIssue(Issue issue, bool getAssignedUsers,
             bool getConversations,
             bool getTimeSheets, bool getParent, bool getPredecessors, bool getSuccessors, bool getAll)
         {
             var issueDto = _issueService.Get(issue.Id);
-            var assignedUsers = getAll || getAssignedUsers
-                ? Task.WhenAll(issue.AssignedUserIds.Select(_userRepository.GetAsync))
-                : null;
-            var conversations = getAll || getConversations
-                ? _conversationService.GetConversationsFromIssueAsync(issue.Id.ToString())
-                : null;
-            var timeSheets = getAll || getTimeSheets
-                ? _timeSheetService.GetAllOfIssueAsync(issue.Id)
-                : null;
-            var parent = getAll || getParent
-                ? _issueService.GetParent(issue.Id)
-                : null;
-            var predecessors = getAll || getPredecessors
-                ? Task.WhenAll(issue.PredecessorIssueIds.Select(_issueService.Get))
-                : null;
-            var successors = getAll || getSuccessors
-                ? Task.WhenAll(issue.SuccessorIssueIds.Select(_issueService.Get))
-                : null;
+            var assignedUsers = getAll || getAssignedUsers ? Task.WhenAll(issue.AssignedUserIds.Select(_userRepository.GetAsync)) : null;
+            var conversations = getAll || getConversations ? _conversationService.GetConversationsFromIssueAsync(issue.Id.ToString()) : null;
+            var timeSheets = getAll || getTimeSheets ? _timeSheetService.GetAllOfIssueAsync(issue.Id) : null;
+            var parent = getAll || getParent ? _issueService.GetParent(issue.Id) : null;
+            var predecessors = getAll || getPredecessors ? Task.WhenAll(issue.PredecessorIssueIds.Select(_issueService.Get)) : null;
+            var successors = getAll || getSuccessors ? Task.WhenAll(issue.SuccessorIssueIds.Select(_issueService.Get)) : null;
 
-            return new IssueDTODetailed((await issueDto).State, (await issueDto).Project, (await issueDto).Client,
-                (await issueDto).Author,
-                assignedUsers != null ? (await assignedUsers).Select(it => new UserDTO(it)).ToList() : null,
+            return new IssueDTODetailed((await issueDto).Id, (await issueDto).State, (await issueDto).Project, (await issueDto).Client,
+                (await issueDto).Author, assignedUsers != null ? (await assignedUsers).Select(it => new UserDTO(it)).ToList() : null,
                 conversations != null ? await conversations : null, timeSheets != null ? await timeSheets : null,
                 issue.IssueDetail, parent != null ? await parent : null,
-                predecessors != null ? await predecessors : null,
-                successors != null ? await successors : null);
+                predecessors != null ? await predecessors : null, successors != null ? await successors : null);
```

Comment:
Du könntest z.B bei den predecessors das await schon in z. 81 setzen und dir dann den ternären ausdruck in z.88 sparen